hal/halx86/legacy/bussupp.c

```diff
@@ -854,36 +854,58 @@ HalpDebugPciDumpBus(IN ULONG i,
     if (VendorName)
     {
         /* Copy the vendor name into our buffer */
-        VendorName += 8;
+        VendorName += strlen("\r\n0000  ");
         p = strpbrk(VendorName, "\r\n");
         Length = p - VendorName;
         if (Length >= sizeof(bVendorName)) Length = sizeof(bVendorName) - 1;
         strncpy(bVendorName, VendorName, Length);
         bVendorName[Length] = '\0';
+        p += strlen("\r\n");
+        while (*p == '\t' || *p == '#')
+        {
+            p = strpbrk(p, "\r\n");
```

Comment:
Is it possible that strpbrk returns NULL in this condition?

Reply:
It's only possible if the PCI database file is broken. :slightly_smiling_face: In details:

If condition `if (VendorName)` is true, we found a line in file. Each line ends with `\r\n`. Let's assume we are at the last line. `p += strlen("\r\n");` will make variable point to the end of file, in our case entire file is a C string, so `*p` will be equal to 0, and loop condition `while (*p == '\t' || *p == '#')` will evaluate to false, thus `strpbrk` is not executed.

Reply:
ok so we expect that there is a trailing newline in the file.